engine/classes/Elgg/Database/Seeds/Seed.php

```diff
@@ -13,17 +13,88 @@
 abstract class Seed implements Seedable {
 
 	use Seeding;
-	use Progressing;
+	use Progressing {
+		advance as private progressAdvance;
+	}
+	
+	/**
+	 * @var int Max number of items to be created by the seed
+	 */
+	protected $limit = 20;
 
+	/**
+	 * @var bool Create new entities
+	 */
+	protected $create = false;
+	
+	/**
+	 * @var int Number of seeded entities
+	 */
+	protected $seeded_counter = 0;
+	
 	/**
 	 * Seed constructor.
 	 *
-	 * @param int $limit Number of item to seed
+	 * @param array $options seeding options
+	 *                       - limit: Number of item to seed
+	 *                       - create: create new entities (default: false)
+	 *                       - create_since: lower bound creation time (default: now)
+	 *                       - create_until: upper bound creation time (default: now)
 	 */
-	public function __construct($limit = null) {
-		if (isset($limit)) {
+	public function __construct(array $options = []) {
+		$limit = (int) elgg_extract('limit', $options);
+		if ($limit > 0) {
 			$this->limit = $limit;
 		}
+		
+		$this->create = (bool) elgg_extract('create', $options, $this->create);
+		$this->setCreateSince(elgg_extract('create_since', $options, 'now'));
+		$this->setCreateUntil(elgg_extract('create_until', $options, 'now'));
+	}
+	
+	/**
+	 * Register this class for seeding
+	 *
+	 * @param \Elgg\Hook $hook 'seeds', 'database'
+	 *
+	 * @return array
+	 */
+	final public static function register(\Elgg\Hook $hook) {
+		$seeds = $hook->getValue();
+		
+		$seeds[] = static::class;
+		
+		return $seeds;
+	}
+	
+	/**
+	 * Get the count of the seeded entities
+	 *
+	 * @return int
+	 */
+	final public function getCount() : int {
+		if ($this->create) {
+			return $this->seeded_counter;
+		}
+		
+		$defaults = [
+			'metadata_names' => '__faker',
+		];
+		$options = array_merge($defaults, $this->getCountOptions());
+		
+		return elgg_count_entities($options);
+	}
+	
+	/**
+	 * Advance progressbar
+	 *
+	 * @param int $step Step
+	 * @return void
+	 */
+	public function advance($step = 1) {
```

Comment:
typehint to int... no need to cast on next line

Reply:
done